### PR TITLE
Avoid edge case in _adjust_learning_rate.

### PR DESCRIPTION
**Pull request by tdomhan**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sockeye/training.py [hunk 1]
```diff
@@ -527,7 +527,8 @@ def _adjust_learning_rate(self, has_improved: bool):
                             self.state.best_checkpoint)
                 adjusted_lr = self.trainer.optimizer.lr_scheduler.lr
                 # trainer.load_states also reloads the parameters
-                self._load_trainer_states(self.best_optimizer_states_fname)
+                if os.path.exists(self.best_optimizer_states_fname):
+                    self._load_trainer_states(self.best_optimizer_states_fname)
```
Comment (tdomhan) [inline sockeye/training.py:531]: `self.best_optimizer_states_fname` only gets written to if `has_improved` is true in `fit()`. For the edge case where has_improved has never been True before the first time the learning rate is reduced training would crash as the file is not found..

Comment (fhieber) [reply sockeye/training.py:531]: nice catch

**Review by fhieber — APPROVED** [review]
